src/node_watcher.js

```diff
@@ -270,16 +270,16 @@ NodeWatcher.prototype.processChange = function(dir, event, file) {
   var fullPath = path.join(dir, file);
   var relativePath = path.join(path.relative(this.root, dir), file);
   fs.lstat(fullPath, function(error, stat) {
+    var registered = this.registered(fullPath);
     if (error && error.code !== 'ENOENT') {
       this.emit('error', error);
     } else if (!error && stat.isDirectory()) {
       // win32 emits usless change events on dirs.
-      if (event !== 'change') {
+      if (event !== 'change' && registered) {
```

Comment:
This will make us exclude other good directories too. `this.registered` will check if we had already registered the directory, so by definition new directories will not be registered. 

We should still watch all new directories. The reason being is that we won't know if files under that directory will come to be significant in the future. Take the following example:

```
var glob = 'a/*.js';
var directory = 'a';
var file = 'a/a.js';

minimatch(directory, glob); // false
minimatch(file, glob); // true
```

However, we shouldn't emit an event unless the directory matches our critiria. So let's put the add event below under the following condition:
```
 common.isFileIncluded(this.globs, this.dot, relativePath)
```